contract/src/lib.rs

```diff
@@ -290,12 +293,33 @@ impl MpcContract {
     }
 
     #[allow(unused_variables)]
-    pub fn sign(&mut self, payload: [u8; 32], path: String) -> [u8; 32] {
-        near_sdk::env::random_seed_array()
+    pub fn sign(&mut self, payload: [u8; 32], path: String) -> Promise {
+        self.pending_requests.insert(&payload, &None);
+        env::log_str(&serde_json::to_string(&near_sdk::env::random_seed_array()).unwrap());
+        Self::ext(env::current_account_id()).sign_helper(payload)
     }
 
-    #[allow(unused_variables)]
-    pub fn respond(&mut self, receipt_id: [u8; 32], big_r: String, s: String) {}
+    pub fn sign_helper(&mut self, payload: [u8; 32]) -> PromiseOrValue<(String, String)> {
```

Comment:
it's gonna panic if called directly, but we should probably add `[private]` as well just in case. I also see a case where someone else also sees this payload and tries to front run the callback